### ISSUE TITLE
Support for Legacy UI features like FrameSequenceTracker and PipelineReporter

### ISSUE THREAD

**Issue by SamFort-MSFT** [google/perfetto#80]
The ui.perfetto.dev site doesn't appear to include support for some of the legacy UI features such as FrameSequenceTracker, PipelineReporter, or InputLatency::*. What is the plan for enabling these features in the new perfetto UI?

**Comment by primiano** [comment]
Can you clarify with some more detail what they are? Not even sure what you are talking about  (some of the things you mentioend can mean too many things, like InputLatency::*).
The plan is not to carry over all the features as-is. THere's a long tail of features that likely will stay only in the "open with legacy ui" path.

**Comment by primiano** [comment]
Expanding a bit more, there is a plan for the UI to become more modular and allow people to create extensions / plugins to deal with these long tail use cases like this.

The failure mode of the legacy ui is trying to have all the possible feature, which makes it a project impossible to run and maintain, and doomed to accumulate unmaintained code written by people who just one some feature and then run away. 
We decided to start this journey (modularizing the perfetto UI) as we were slowly creeping into the "let's re-add all the legacy features" game, which would reintroduce all the problems.

Unfortunately re-architecting the UI, while being a priority for us, is also going to be a massive task and will take a while (don't have a solid estimate right now as we are just getting started, but somewhere in the 1y ballpark)

**Comment by SamFort-MSFT** [comment]
Thank you for the additional context.

InputLatency is a tracing visualization of LatencyInfo (chromium.src's ui/latency/latency_info.h) that tracks input event timings as they work through the various stages in the pipeline. They are exposed in the legacy UI under the browser process and let one quickly identify slow input event processing for further analysis. For example, I use it regularly to understand scrolling performance.

FrameSequenceTracker lets one more easily understand Chromium's ability to produce frames in response to a frame generating action such as a scroll. For example, it lets one easily see if a scroll produced only 93 of the expected 100 frames of animation along with which frames were slower or missed. This is then leveraged to dig deeper into why those specific frames were slow.

PipelineReporter lets one visualize (with timing information) the steps that a frame goes through both processes and in parallel with other frames as it moves from viz/gpu to renderer to viz/gpu before ultimately being presented on screen. This is frequently used in parallel with FrameSequenceTracker to understand why frames weren't produced at the desired rate.

The most recent comment about modularizing the UI and supporting an extension/plugin model makes a lot of sense. I'm not surprised that the legacy UI has acquired a lot of stale functionality for the reasons you mention.

Going forward, if there is a modularized UI that supports plugins/extensions in ~1y, what is the roadmap for the existing functionality that the legacy UI does provide? Is it something like:

- UI modularization and extensibility system put into place in ~1y (using your ballpark estimate)
- Teams are given some bound timeframe to re-write their functionality using extensibility mechanism
- Legacy UI continues to be supported for that timeframe before being removed

Or is there another plan?

**Comment by primiano** [comment]
In my mind:
1. The legacy ui button won't go away from the "open as legacy ui" for a long while (even past the y)
2. It's going to be up to teams to create equivalent of new features once the modularization is there. I suppose that popular features like frame tracer and the like might be ported, other small ones that nobody / few people care about will only live in the legacy ui.

it really depends by how many people want that and if they are willing to contribute.

**Comment by primiano** [comment]
@deepanjanroy in case he has opinions

**Comment by deepanjanroy** [comment]
I agree with primiano@. These features will only be available in the legacy UI until 
(a) Perfetto UI is more extensible and
(b) someone is willing to write the extension. 

(a) is the harder part. Once (a) is done, each individual feature likely won't be too difficult to implement.

**Comment by SamFort-MSFT** [comment]
Thank you for the additional information. Sounds good and I'll look forward to the Perfetto UI extensibility work landing!
